### PR TITLE
[PoC] ci: avoid 'brew' in the OSX CI jobs

### PR DESCRIPTION
**Pull request by szeder**
Over the years we had a good couple of issues with Homebrew that
resulted in broken OSX builds.  Maybe it's worth a try to manage the
build dependencies ourselves, and see what happens.  More details in
the second commit.

This is just a proof of concept, to satisfy my curiosity; I'm not sure
whether it's a good idea or just about the dumbest idea ever.  And
would be quite surprised if it worked on Azure Pipelines...

### CODE REVIEW

**Comment by szeder** [comment]
> And would be quite surprised if it worked on Azure Pipelines...

No surprises, it doesn't work.

**Comment by dscho** [comment]
@szeder sorry, I had to restart a bunch of builds because of the `perforce` package being outdated. In related news: I managed to teach the Azure Pipeline to automatically open a PR and it was already merged: https://github.com/Homebrew/homebrew-cask/pull/71990

**Comment by dscho** [comment]
@szeder I restarted the job that had timed out, so the build passed now.

**Comment by dscho** [comment]
FWIW with https://github.com/Homebrew/homebrew-cask/pull/75988 being open already for 17 hours without any indication when it might get merged, I am starting to turn around on my earlier assessment, coming around to your point of view: avoiding to rely on Homebrew so much might be our best bet in Git's CI/PR builds.

I also like 6d9461b7f229e34108daceaf7553f52a7cf9282f, I have to admit. Much cleaner that way.

File: ci/install-dependencies.sh [hunk 1]
```diff
@@ -34,26 +34,16 @@ linux-clang|linux-gcc)
...
-		# Just in case the image is updated to contain gcc@9
-		# pre-installed but not linked.
-		brew link gcc@9
-		;;
-	esac
+	(
+		cd ~
+		ver=$(sw_vers | sed -n -e 's/^ProductVersion:[[:space:]]*\([0-9]*\.[0-9]*\)\..*/\1/p')
+		git clone --branch=ci-deps-osx-$ver --single-branch --depth=1 \
+			https://github.com/szeder/git deps
```
Comment (dscho) [inline ci/install-dependencies.sh]: Dunno. I'd rather use Azure Build Artifacts and a scheduled build that keeps it up to date...

**Pull request closed without merging**